processor/src/test/java/org/mapstruct/ap/testutil/runner/GeneratedSource.java

```diff
@@ -39,11 +39,21 @@
  */
 public class GeneratedSource implements TestRule {
 
+    private static ThreadLocal<CompilingStatement> compilingStatement = new ThreadLocal<CompilingStatement>();;
```

Comment:
Before the change, there was a lot of ThreadLocal stuff in the CompilingStatement itself. I've removed all of that over there, but this part remains, as a `TestRule` is not by API limited to be used as a `@Rule`, but can also be used as `@ClassRule`. If it was used that way and if the test methods would be executed in parallel, then we could run into troubles here.
As I'm writing this and checking in the code, I just notice that the usage of the `CompilationCache` is not thread-safe for parallel method execution anymore... uhm...
I guess I'll need to think about that. I guess it makes sense to just ditch the support of parellel method execution altogether (would kill the cache effect anyway), and just support the parallel class execution.